opensearch-observability/src/test/kotlin/org/opensearch/observability/bwc/TABackwardCompatibilityIT.kt

```diff
@@ -0,0 +1,128 @@
+                    "upgraded_cluster" -> UPGRADED
+                    else -> {
+                        throw AssertionError("unknown cluster type: $value")
+                    }
+                }
+            }
+        }
+    }
+
+    @Throws(Exception::class)
+    @SuppressWarnings("unchecked")
+    fun `test backwards compatibility`() {
+        val uri = getUri()
+        val responseMap = getAsMap(uri)["nodes"] as Map<String, Map<String, Any>>
+        for (response in responseMap.values()) {
+            val plugins = response["plugins"] as List<Map<String, Any>>
+            val pluginNames = plugins.map { plugin -> plugin["name"] }.toSet()
+            return when (CLUSTER_TYPE) {
+                ClusterType.OLD -> {
+                    assertTrue(pluginNames.contains("opensearch-notebooks"))
+                    createNotebook()
+                }
+                ClusterType.MIXED -> {
+                    assertTrue(pluginNames.contains("opensearch-observability"))
+                    if (System.getProperty("tests.rest.bwcsuite_round") != "third") {
+                        verifyNotebooksExists("$BASE_NOTEBOOKS_URI/notebooks")
+                    } else verifyNotebooksExists("$BASE_OBSERVABILITY_URI/object")
+                }
+                ClusterType.UPGRADED -> {
+                    assertTrue(pluginNames.contains("opensearch-observability"))
+                    verifyNotebooksExists("$BASE_OBSERVABILITY_URI/object")
+                }
+            }
+            break
+        }
+    }
+
+    private fun getUri(): String {
+        return when (CLUSTER_TYPE) {
+            ClusterType.OLD -> "_nodes/" + CLUSTER_NAME + "-0/plugins"
+            ClusterType.MIXED -> {
+                when (System.getProperty("tests.rest.bwcsuite_round")) {
+                    "second" -> "_nodes/$CLUSTER_NAME-1/plugins"
+                    "third" -> "_nodes/$CLUSTER_NAME-2/plugins"
+                    else -> "_nodes/$CLUSTER_NAME-0/plugins"
+                }
+            }
+            ClusterType.UPGRADED -> "_nodes/plugins"
+        }
+    }
+
+    private fun createNotebook() {
+        val createRequest = constructNotebookRequest()
+        val createResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_NOTEBOOKS_URI/notebook",
+            createRequest,
+            RestStatus.OK.status
+        )
+        val id = createResponse.get("notebookId").asString
+        Assert.assertNotNull("Id should be generated", id)
+        Thread.sleep(100)
+    }
+
+    private fun verifyNotebooksExists(uri: String) {
+        val listNotebooks = executeRequest(
+            RestRequest.Method.GET.name,
+            "$uri",
+            "",
+            RestStatus.OK.status
+        )
+        val totalHits = listNotebooks.get("totalHits").asInt
+        assertTrue("Actual notebooks counts ($totalHits) should be greater than or equal to (1)", totalHits >= 1)
```

Comment:
why will it be > 1?

Reply:
I'm checking if the notebooks created in 1.1 version can be retrieved after the version upgrade too, thats why its > 1

Reply:
but you only created 1 notebook in 1.1?

Reply:
ohh yes, I can change it to just == 1